presto-druid/src/main/java/io/prestosql/plugin/druid/DruidJdbcClient.java

```diff
@@ -0,0 +1,228 @@
+        }
+        return Optional.of(varcharColumnMapping(createVarcharType(columnSize)));
+    }
+
+    @Override
+    public List<JdbcColumnHandle> getColumns(ConnectorSession session, JdbcTableHandle tableHandle)
+    {
+        try (Connection connection = connectionFactory.openConnection(JdbcIdentity.from(session))) {
+            try (ResultSet resultSet = getDruidColumns(tableHandle, connection.getMetaData())) {
+                List<JdbcColumnHandle> columns = new ArrayList<>();
+                while (resultSet.next()) {
+                    JdbcTypeHandle typeHandle = new JdbcTypeHandle(
+                            resultSet.getInt("DATA_TYPE"),
+                            // type_name isn't supported in Druid
+                            Optional.of(resultSet.getString("TYPE_NAME")),
+                            resultSet.getInt("COLUMN_SIZE"),
+                            resultSet.getInt("DECIMAL_DIGITS"),
+                            // array_dimensions isn't supported in Druid
+                            Optional.ofNullable(null));
+                    Optional<ColumnMapping> columnMapping = toPrestoType(session, connection, typeHandle);
+                    // skip unsupported column types
+                    if (columnMapping.isPresent()) {
+                        String columnName = resultSet.getString("COLUMN_NAME");
+                        boolean nullable = (resultSet.getInt("NULLABLE") != columnNoNulls);
+                        columns.add(new JdbcColumnHandle(columnName, typeHandle, columnMapping.get().getType(), nullable));
+                    }
+                }
+                if (columns.isEmpty()) {
+                    // In rare cases (e.g. PostgreSQL) a table might have no columns.
+                    throw new TableNotFoundException(tableHandle.getSchemaTableName());
+                }
+                return ImmutableList.copyOf(columns);
+            }
+        }
+        catch (SQLException e) {
+            throw new PrestoException(JDBC_ERROR, e);
+        }
+    }
+
+    /**
+     * Custom way of retrieving column names since the {@link BaseJdbcClient#getColumns(JdbcTableHandle, DatabaseMetaData)}
+     * method uses character escaping that doesn't jive well with Druid SQL.
+     */
+    private static ResultSet getDruidColumns(JdbcTableHandle tableHandle, DatabaseMetaData metadata)
+            throws SQLException
+    {
+        return metadata.getColumns(
+                tableHandle.getCatalogName(),
+                tableHandle.getSchemaName(),
+                tableHandle.getTableName(),
+                null);
+    }
+
+    /**
+     * Overriding this method to handle following weirdness/issues in Druid:
+     * 1) Druid doesn't like table names to be qualified with catalog names in the SQL query. So setting it to an empty string.
+     * 2) Need to use DruidQueryBuilder since Druid doesn't like SELECT NULL queries when there are no columns projected.
```

Comment:
We can modify the `QueryBuilder` to eg `SELECT 1`. Would that help?

Reply:
This is exactly what we are doing right now in `DruidQueryBuilder`

```
@Override
protected String getProjection(List<JdbcColumnHandle> columns)
{
        // Druid doesn't support SELECT NULL queries
        if (columns.isEmpty()) {
            return "1";
        }
        return super.getProjection(columns);
}
```

Reply:
perfect; then let's change the original class to avoid making a copy/subclass